### PR TITLE
Convert from Mapbox GL to MapLibre

### LINKED ISSUE #781: Support for maplibre
Since this topic was raised in https://github.com/maputnik/editor/issues/775, this issue is to discuss about support for maplibre: https://github.com/maplibre/maplibre-gl-js

- mapbox-gl version 1.x. -> backward compatible to mapbox-gl

- maplibre-gl version 2 -> non-backward compatible

### PR DESCRIPTION
**Pull request by kevinschaul**
Picking up #789 where @lseelenbinder left it. The tests are now passing for me locally, and things seem to be working in the limited testing I've done manually.

Fixes #781.

### CODE REVIEW

**Comment by lseelenbinder** [comment]
Thanks so much @kevinschaul! I'll take a look tomorrow. Super glad someone was able to pick up where I left off. :smile:

**Comment by lseelenbinder** [comment]
I think we can probably drop tests and build for Node 12, 14, and 16 in favor of 18? Any objections?

**Comment by kevinschaul** [comment]
Great, thanks @lseelenbinder!

Looks like build/artifacts is failing to install the maputnik desktop version. I had trouble installing the desktop version in January but solved it with this PR: https://github.com/maputnik/desktop/pull/17 I wonder if merging that would resolve this build issue.

**Comment by nyurik** [comment]
@kevinschaul I think we should merge that PR first, although I don't see it passing any CI... perhaps because I missed it initially.  That PR seems reasonable, but I'm not a GO expert

**Comment by lseelenbinder** [comment]
The go part looks good to me @kevinschaul, as well. (Also not an expert, but I've done a fair share of work with it.)

**Review by lseelenbinder — APPROVED** [review]

**Comment by lseelenbinder** [comment]
PR looks good. I'd like to see more passing CI, but that may be better to do on `master` once it's merged and we're in a workable state.

**Comment by kevinschaul** [comment]
Apologies, I shared an older PR to maputnik/desktop. This PR fixes CI too. I think this is the better candidate to merge: https://github.com/maputnik/desktop/pull/18

**Review by nyurik — APPROVED** [review]
looks awesome!  Is this ready to merge?

**Comment by nyurik** [comment]
CI/codesandbox is failing, and I suspect it can be solved with one of the ways to accept the host, e.g. `-o "StrictHostKeyChecking no"` or similar.  Not sure if it should be done as part of this PR though

**Comment by nyurik** [comment]
Merged!  I agree that 2.x is good for this kind of change :)

File: src/mapboxgl.css [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* TODO */
```
Comment (tordans) [inline src/mapboxgl.css:1]: This might be worth moving to a separate / new issue, describing the TODO in more detail.

**Pull request merged**